compaction/compaction_manager.cc

```diff
@@ -962,7 +962,14 @@ compaction_manager::compaction_manager(config cfg, abort_source& as, tasks::task
     , _update_compaction_static_shares_action([this] { return update_static_shares(static_shares()); })
     , _compaction_static_shares_observer(_cfg.static_shares.observe(_update_compaction_static_shares_action.make_observer()))
     , _strategy_control(std::make_unique<strategy_control>(*this))
-    , _tombstone_gc_state(&_reconcile_history_maps) {
+    , _tombstone_gc_state(&_reconcile_history_maps)
+    , _out_of_space_subscription(oos_controller && (this_shard_id() == 0) ? oos_controller->subscribe([this] (auto critical_level_reached) {
+        if (bool(critical_level_reached)) {
+            return container().invoke_on_all([] (compaction_manager& cm) { return cm.drain(); });
+        }
+        return container().invoke_on_all([] (compaction_manager& cm) { cm.enable(); });
```

Comment:
Shouldn't `enable()` or `drain()` be called only if `state == running`?
Otherwise, if the callback is called early before the compaction manager was enabled the first time there is no need to drain() it, and enable() would be premature as well.